### PR TITLE
Add basic testing automation against all version of Python using nox

### PR DESCRIPTION
**Pull request by abey79**
### What

This PR introduces basic support to run tests and examples against all supported version of Python (currently 3.8 to 3.11). Currently, running tests and `run_all.py` is supported.

Usage:
```
nox --help
nox -s tests
nox -s run_all -- --save
```

For discoverability and convenience, I added a couple of `just` commands:

```
just py-tests-allpy
just py-run-all-allpy --save
```

On the way, I added the capability to install example requirements to `run_all.py`:

```
python scripts/run_all.py --install-requirements
```

Decisions:
- I used [nox](https://nox.thea.codes/en/stable/). The alternative is the older,  more widespread [tox](https://tox.wiki/en/latest/). Reasons: `noxfile.py` is an actual Python script (as are most of our scripts already) and `nox` is battle-tested enough to be used in some large projects, including pip itself.
- Nox is best installed globally on the dev machine (as it manages its own venvs for running stuff). I updated `setup_dev.sh` accordingly, by installing `pipx` using the OS package manager, and then using pipx to install nox. (This is The Right Way™️ to globally install Python CLIs.)

Note: this requires all version of python to be installed and available on the dev machine. Can't recommend MacPorts enough: `sudo port install  python38 python39 python310 python311`.

Relates to:
- #2445

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)

PR Build Summary: https://build.rerun.io/pr/2536

Docs preview: https://rerun.io/preview/2d89edf/docs
Examples preview: https://rerun.io/preview/2d89edf/examples

### CODE REVIEW

File: noxfile.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+from __future__ import annotations
```
Comment (emilk) [inline noxfile.py:1]: please add a description of this file (what it is for, how it is used, …)

File: noxfile.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    session.install("-r", "rerun_py/requirements-build.txt")
+    session.install("./rerun_py")
+    session.run("just", "py-test")
+
+
+@nox.session(python=["3.8", "3.9", "3.10", "3.11"])
+def examples(session: nox.Session) -> None:
+    session.install("-r", "examples/python/requirements.txt")
+    session.install("./rerun_py")
+    session.run("python", "scripts/run_all.py", "--save")
```
Comment (emilk) [inline noxfile.py:17]: could this file be moved to e.g. scripts/ so as not to pollute the repo root further?

**Comment by emilk** [comment]
I ran `brew install pipx` then:

```
❯ pipx install nox
  installed package nox 2023.4.22, installed using Python 3.11.4
  These apps are now globally available
    - nox
    - tox-to-nox
⚠️  Note: '/Users/emilk/.local/bin' is not on your PATH environment
    variable. These apps will not be globally accessible until your
    PATH is updated. Run `pipx ensurepath` to automatically add it,
    or manually modify your PATH in your shell's config file (i.e.
    ~/.bashrc).
done! ✨ 
```

yet:

```
❯ nox -s examples
zsh: command not found: nox
```

so I guess there is some more step that is required

File: noxfile.py (hunk 2 above)
Comment (abey79) [reply noxfile.py:17]: Yes, but that would mean running `nox -f scripts/noxfile.py -s tests`. I guess that can be hidden behind `just nox -s tests`. Prefer that way?

**Comment by abey79** [comment]
> I ran `brew install pipx` then:
> 
> ```
> ❯ pipx install nox
>   installed package nox 2023.4.22, installed using Python 3.11.4
>   These apps are now globally available
>     - nox
>     - tox-to-nox
> ⚠️  Note: '/Users/emilk/.local/bin' is not on your PATH environment
>     variable. These apps will not be globally accessible until your
>     PATH is updated. Run `pipx ensurepath` to automatically add it,
>     or manually modify your PATH in your shell's config file (i.e.
>     ~/.bashrc).
> done! ✨ 
> ```
> 
> yet:
> 
> ```
> ❯ nox -s examples
> zsh: command not found: nox
> ```
> 
> so I guess there is some more step that is required

That's right, I forgot about this. As noted, '/Users/emilk/.local/bin' must be added to the path. Luckily, pix has support for that `pipx ensure path`. I'll add it to `setup_dev.sh`.

File: noxfile.py (hunk 2 above)
Comment (abey79) [reply noxfile.py:24]: I just looked into it, and I'm a bit split. It's not tooo bad to move it and force using `just nox ...`, but does generate some friction wrt usual nox expectations, incl:
- one must change the current directory to `..` in each session function
- the `.nox` cache dir weirdly ends up in `scripts/` as well

Also, I believe nox could help with #2200, so I'm inclined to give it first-class citizen status.

Comment (emilk) [reply noxfile.py:24]: ok!

**Comment by emilk** [comment]
Please add the command to `justfile` with some descriptive name. No-one will know that nox is even a thing if we don't put it somewhere discoverable

**Comment by abey79** [comment]
@emilk Done, plus a bunch of related improvements.

File: justfile [hunk 3]
```diff
@@ -45,6 +45,10 @@ py-run-all-web:
 py-run-all-rrd *ARGS:
     just py-run-all --save {{ARGS}}
 
+# Run all example with all supported Python versions (through nox)
```
Comment (emilk) [inline justfile:48]:
```suggestion
# Run all examples with all supported Python versions (through nox)
```

**Review by emilk — APPROVED** [review]
Great!

**Pull request merged**